### PR TITLE
Feature/2806 add clear wishlist button

### LINKED ISSUE #2806: Add 'clear wishlist' button.
## What is the motivation for adding / enhancing this feature?
Similar to clear cart button - currently if user has many products in their wishlist, and want to start fresh, he needs to remove them one by one. Possibility to clear whole wishlist with two clicks (click button + confirmation) would be nice improvement. Adding product to wishlist means less commitment that adding them to cart - so many users tend to add a lot of products to their wishlists. 

## What are the acceptance criteria 
- [ ] - New 'Clear wishlist' Button is displayed in the upper right part of wishlist sidebar
- [ ] - Location and button design should be identical as 'Clear cart' button introduced in #2587
- [ ] - Clicking this button should result in displaying confirmation popup, and after confirmation - clearing wishlist. 

## Can you complete this feature request by yourself?

- [ ] YES
- [x] NO

## Which [Release Cycle](https://docs.vuestorefront.io/guide/basics/release-cycle.html) state this refers to? Info for developer.
Pick one option.

- [x] This is a normal feature request. This should be available on https://test.storefrontcloud.io and then after tests this can be added to next Vue Storefront version. In this case Developer should create branch from `develop` branch and create Pull Request `2. Feature / Improvement` back to `develop`.
- [ ] (**Pick this option only if you're sure**) This is an important improvement request for current Release Candidate version on https://next.storefrontcloud.io and should be placed in next RC version. In this case Developer should create branch from `release` branch and create Pull Request `3. Stabilisation fix` back to `release`.
- [ ] (**Pick this option only if you're sure**) This is a critical improvement request for current Stable version on https://demo.storefrontcloud.io and should be placed in next stable version. In this case Developer should create branch from `hotfix` or `master` branch and create Pull Request `4. Hotfix` back to `hotfix`.

## Additional information
Adding and moving 'clear cart' button to the correct location, should be a good example of how to proceeed:

#2587
#2765

### PR DESCRIPTION
**Pull request by aniamusial**
### Related issues

closes #2806 

### Short description and why it's useful
It adds the "Clear wishlist" button and it gets the job done in one click, instead of many :)

### Screenshots of visual changes before/after (if there are any)
Screenshot from 2019-05-20 11-08-33
Screenshot from 2019-05-20 11-08-20

### Which environment this relates to

- [x] Test version (https://test.storefrontcloud.io) - this is a new feature or improvement for Vue Storefront. I've created branch from `develop` branch and want to merge it back to `develop`
- [ ] RC version (https://next.storefrontcloud.io) - this is a stabilisation fix for Release Candidate of Vue Storefront. I've created branch from `release` branch and want to merge it back to `release`
- [ ] Stable version (https://demo.storefrontcloud.io) - this is an important fix for current stable version. I've created branch from `hotfix` or `master` branch and want to merge it back to `hotfix`

### Upgrade Notes and Changelog

- [x] No upgrade steps required (100% backward compatibility and no breaking changes)
- [ ] I've updated the [Upgrade notes](https://github.com/DivanteLtd/vue-storefront/blob/develop/doc/Upgrade%20notes.md) and [Changelog](https://github.com/DivanteLtd/vue-storefront/blob/develop/CHANGELOG.md) on how to port existing VS sites with this new feature

### Contribution and currently important rules acceptance

- [x] I read and followed [contribution rules](https://github.com/DivanteLtd/vue-storefront/blob/master/CONTRIBUTING.md)
- [x] I read the [TypeScript Action Plan](https://github.com/DivanteLtd/vue-storefront/blob/master/doc/TypeScript%20Action%20Plan.md) and adjusted my PR according to it
- [x] I read about [Vue Storefront Modules](https://github.com/DivanteLtd/vue-storefront/blob/master/doc/api-modules/about-modules.md) and I am aware that every new feature should be a module

### CODE REVIEW

File: src/themes/default/components/core/blocks/Wishlist/ClearWishlistButton.vue [hunk 1]
```diff
@@ -0,0 +1,19 @@
+<template>
+  <button class="brdr-none bg-cl-transparent p0 middle-xs inline-flex cl-secondary weight-400 h4 sans-serif fs-medium">
```
Comment (patzick) [inline src/themes/default/components/core/blocks/Wishlist/ClearWishlistButton.vue]: this should emit click event

File: src/themes/default/components/core/blocks/Wishlist/Wishlist.vue [hunk 2]
```diff
@@ -5,9 +5,12 @@
...
       </div>
     </div>
-    <h2 v-if="productsInWishlist.length" class="cl-accent ml30">
-      {{ $t('Wishlist') }}
-    </h2>
+    <div class="row middle-xs px40">
+      <h2 v-if="productsInWishlist.length" class="col-xs-12 col-sm cl-accent">
+        {{ $t('Wishlist') }}
+      </h2>
+      <clear-wishlist-button v-if="productsInWishlist.length" @click.native="clearWishlist" class="col-xs-12 col-sm mt35 mb35 end-sm"/>
```
Comment (patzick) [inline src/themes/default/components/core/blocks/Wishlist/Wishlist.vue]:
```suggestion
      <clear-wishlist-button v-if="productsInWishlist.length" @click="clearWishlist" class="col-xs-12 col-sm mt35 mb35 end-sm"/>
```
we should avoid using `native` - if it's our component button should emit click event

File: src/themes/default/components/core/blocks/Wishlist/Wishlist.vue [hunk 3]
```diff
@@ -27,6 +30,8 @@
 <script>
 import Wishlist from '@vue-storefront/core/compatibility/components/blocks/Wishlist/Wishlist'
 import Product from 'theme/components/core/blocks/Wishlist/Product'
+import ClearWishlistButton from 'theme/components/core/blocks/Wishlist/ClearWishlistButton'
+import i18n from '@vue-storefront/i18n'
```
Comment (patzick) [inline src/themes/default/components/core/blocks/Wishlist/Wishlist.vue]:
```suggestion
```

not needed, we have `this.$t` for that

File: src/themes/default/components/core/blocks/Wishlist/Wishlist.vue [hunk 4]
```diff
@@ -37,7 +42,23 @@ export default {
...
   components: {
-    Product
+    Product,
+    ClearWishlistButton
+  },
+  methods: {
+    clearWishlist () {
+      this.$store.dispatch('notification/spawnNotification', {
+        type: 'warning',
+        message: i18n.t('Are you sure you would like to remove all the items from the wishlist?'),
```
Comment (patzick) [inline src/themes/default/components/core/blocks/Wishlist/Wishlist.vue]:
```suggestion
        message: this.$t('Are you sure you would like to remove all the items from the wishlist?'),
```

File: src/themes/default/components/core/blocks/Wishlist/Wishlist.vue [hunk 5]
```diff
@@ -37,7 +42,23 @@ export default {
...
+        action1: { label: i18n.t('OK'),
```
Comment (patzick) [inline src/themes/default/components/core/blocks/Wishlist/Wishlist.vue]:
```suggestion
        action1: { label: this.$t('OK'),
```

File: src/themes/default/components/core/blocks/Wishlist/Wishlist.vue [hunk 6]
```diff
@@ -37,7 +42,23 @@ export default {
...
+          action: () => {
+            this.$store.dispatch('wishlist/clear')
+          }
+        },
+        action2: { label: i18n.t('Cancel'), action: 'close' },
```
Comment (patzick) [inline src/themes/default/components/core/blocks/Wishlist/Wishlist.vue]:
```suggestion
        action2: { label: this.$t('Cancel'), action: 'close' },
```

File: src/themes/default/resource/i18n/en-US.csv [hunk 7]
```diff
@@ -114,6 +114,7 @@
 "My loyalty card","My loyalty card"
 "My product reviews","My product reviews"
 "Wishlist","Wishlist"
+"Clear wishlist","Clear wishlist"
```
Comment (patzick) [inline src/themes/default/resource/i18n/en-US.csv]: do we have `Are you sure you would like to remove all the items from the wishlist?` text here as well?

Comment (aniamusial) [reply src/themes/default/resource/i18n/en-US.csv]: Now we do ;)

File: src/themes/default/components/core/blocks/Wishlist/Wishlist.vue (hunk 2 above)
Comment (aniamusial) [reply src/themes/default/components/core/blocks/Wishlist/Wishlist.vue]: done

**Pull request merged**